tools/Assembly/GFF/build_assembly_gff.py

```diff
@@ -213,20 +238,27 @@ def build_gff(annotations, faa):
         '-o', dest='out', help='Ouput GFF file name', required=True)
     args = parser.parse_args()
 
-    with open(args.out, 'w', buffering=1) as out_handle:
+    annotations = {}
+    load_annotation(args.egg, EggResult, annotations)
+    load_annotation(args.interpro, InterProResult, annotations)
 
-        print('##gff-version 3', file=out_handle)
-
-        annotations = {}
-        load_annotation(args.egg, EggResult, annotations)
-        load_annotation(args.interpro, InterProResult, annotations)
+    if len(annotations) < 1:
+        print("No annotations loaded, aborting")
+        sys.exit(1)
 
+    records = 0
+    with open(args.out, 'w', buffering=1) as out_handle:
+        print('##gff-version 3', file=out_handle)
         for row in build_gff(annotations, args.faa):
             print('\t'.join(row), file=out_handle)
+            records += 1
+    if records == 0:
+        print("No annotations in GFF, aborting")
+        sys.exit(1)
```

Comment:
My only concern is that the pipeline should not fail if there are not antiSMASH annotations.

@KeteSakharova will this make the pipeline fail if there are no antiSMASH annotations?

Reply:
no, there should be a flag-file 'no_antismash' genearated instead of antismash files

Reply:
ok, I will avoid to terminate the process and create a no_antismash flag-flie

Reply:
I think this check is useless because functional annotation step will run if check_value == number of sequences in FAA file != 0
```
# -----------------------------------  << STEP FUNCTIONAL ANNOTATION >>  -----------------------------------
# GFF
# DIAMOND
# KEGG pathways, move to pathways-systems
# Genome Properties
# tsv -> csv
# move antismash summary to pathways-systems

  functional_annotation_and_post_processing:
    when: $(inputs.check_value != 0)
```

Reply:
The counting is not only for the sequences in FAA, it counts any record written, so, if we have FAA sequence but no annotations, it will activate the no_antismash flag). This was the main cause of the error annotating LR assemblies, as the FAA headers were from FGS, the original script was unable to parse those and skip all annotations, generating a GFF with the header line but no data (a non-empty file and the script didn't raise an error).

To avoid this again, the final checkup will check if any record was written, if no records, the no_antismash flag will be generated and the GFF deleted, skipping the steps for sorting, bgziping and indexing.

Reply:
OK, be attentive with final location of no_antismash file then. Because webuploader wants it to be under pathways-systems folder (I guess, better check).